test/e2e/packetcapture_test.go

```diff
@@ -548,14 +666,20 @@ func runPacketCaptureTest(t *testing.T, data *TestData, tc pcTestCase) {
 				t.Logf("Ping(%s) '%s' -> '%v' failed: ERROR (%v)", protocol.StrVal, srcPod, *dstPodIPs, err)
 			}
 		} else if protocol == tcpProto {
-			for i := 1; i <= 10; i++ {
-				if err := data.runNetcatCommandFromTestPodWithProtocol(srcPod, data.testNamespace, toolboxContainerName, server, serverPodPort, "tcp"); err != nil {
+			if tc.pc.Spec.Direction == crdv1alpha1.CaptureDirectionBoth {
+				if err := data.runNetcatCommandFromTestPodWithProtocol(srcPod, data.testNamespace, toolboxContainerName, server, serverPodPort, "tcp", 1); err != nil {
 					t.Logf("Netcat(TCP) '%s' -> '%v' failed: ERROR (%v)", srcPod, server, err)
 				}
+			} else {
+				for i := 1; i <= 10; i++ {
```

Comment:
for me this is not ok and we need a better approach, even as part of this PR
I don't see the logic for having a single retry for the both direction case, while combining 2 retry mechanisms (a loop with 10 iterations + the 5 "retries" argument to `runNetcatCommandFromTestPodWithProtocol`) for the single-direction case.
adding @hangyan in case he has suggestions